model/task/test_result.go

```diff
@@ -345,3 +360,62 @@ func sortTestResults(results []testresult.TestResult, opts *FilterOptions, baseS
 		return false
 	})
 }
+
+func (o TestResultOutput) GetPrestoBucket(ctx context.Context, credentials evergreen.S3Credentials) (pail.Bucket, error) {
```

Comment:
Is there a plan to use this outside of the agent? If not, I'd be inclined to move it into the agent as a helper function to split the agent and app servers more cleanly. Because the agent an app server share the same data models, there's always a long-term maintenance concern that increasing the package dependencies between the two will result in complexity around what's in the model package but is actually used by the agent vs. the app server.

Reply:
The test result service that serves graphQL and our API downloads test results which in turn need this function, so this function is needed both agent and server side.